sql/auth/sql_security_ctx.cc

```diff
@@ -624,6 +624,15 @@ std::pair<bool, bool> Security_context::has_global_grant(const char *priv,
 
   std::string privilege(priv, priv_len);
 
+  if (acl_utility_user.user) {
+    std::string pu(m_priv_user, m_priv_user_length);
+    std::string ph(m_priv_host, m_priv_host_length);
+    if (acl_is_utility_user(pu.c_str(), ph.c_str(), nullptr)) {
```

Comment:
Here, converting to `std::string` just to use `c_str()` then
```suggestion
    if (acl_is_utility_user(m_priv_user, m_priv_host, nullptr)) {
```

Reply:
You are right. I didn't notice that m_priv_user and m_priv_host are assigned only by assign_priv_user/host functions and are null-terminated strings.
Fixed

Reply:
You are right. I didn't notice that m_priv_user and m_priv_host are assigned only by assign_priv_user/host functions and are null-terminated strings.
Fixed